source/codegen/metadata/nixnet/config.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+config = {
+    'api_version': '21.0.0',
+    'c_header': 'nixnet.h',
+    'c_function_prefix': 'nx',
+    'service_class_prefix': 'NiXnet',
+    'java_package': 'com.ni.grpc.xnet',
+    'csharp_namespace': 'NationalInstruments.Grpc.NiXnet',
+    'namespace_component': 'nixnet',
+    'close_function': 'Clear',
+    'code_readiness': 'Release',
+    'driver_name': 'NI-XNET',
+    'status_ok': 'status >= 0',
+    'type_to_grpc_type': {
+        'nxSessionRef_t': 'nidevice_grpc.Session',
+        'nxDatabaseRef_t': 'nidevice_grpc.Session',
+        'nxDatabaseRef_t[]': 'repeated nidevice_grpc.Session',
+        'u32': 'uint32',
+        'u64': 'uint64',
+        'u8': 'uint32',
+        'f64': 'double',
+        'char[]': 'string',
+        'const char[]': 'string',
+        'nxStatus_t': 'int32',
+        'void[]': 'bytes',
+        'nxTimestamp1ns_t': 'uint64',
```

Comment:
This (and probably line 19 above) is causing linux build to fail right now. Timestamps are defined as `u64` in `nixnet.h`. `u64` is [defined as](https://p4.natinst.com/browser/perforce/build/exports/ni/xnet/xnetGlobals/official/export/21.5/21.5.0f186/includes/nixnet.h#line_3157) `unsigned int64` on windows and `unsigned long long int` on linux.

Reply:
Oy, I think there are some force casts for that throughout the codebase. Probably special cases for type == int64/uint64. We might be able to change those to use the grpc_type. Or maybe be more explicit about requiring a force/reinterpret cast in the metadata.

Reply:
@astarche is this change to [service_helpers.mako](https://github.com/ni/grpc-device/pull/570/files#diff-18770917bb2bf21638be75898bfff78d285c52a49af4e2ba348424e57b70f8e0) below is what you had in mind regarding reinterpret cast?

Reply:
Yeah, that's fine. Ideally we want to avoid using driver-specific types in the codegen. Often a good way to do that is to key the logic on the grpc_type. Or to add additional info to the metadata. But, in this case, the existing code is obviously hardocding a bunch of driver types, so adding another is OK. You could also consider a helper method that explains what that is for (different typedef'd integer values with the same guaranteed size but different `long` qualifier counts).